pipeline/build/build.go

```diff
@@ -30,16 +34,45 @@ func (Pipe) Description() string {
 func (Pipe) Run(ctx *context.Context) error {
 	for _, build := range ctx.Config.Builds {
 		log.WithField("build", build).Debug("building")
+		if err := checkMain(ctx, build); err != nil {
+			return err
+		}
 		if err := runPipeOnBuild(ctx, build); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func checkMain(ctx *context.Context, build config.Build) error {
+	var dir = strings.Replace(build.Main, "main.go", "", -1)
```

Comment:
What happens when you pass a file? Or does it only handle directories? You know that the main function doesn't have to be in `main.go`, right?